http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ApacheHttpClient.java

```diff
@@ -304,6 +311,27 @@ default Builder apply(Consumer<Builder> mutator) {
          * Configure the maximum amount of time that a connection should be allowed to remain open while idle.
          */
         Builder connectionMaxIdleTime(Duration maxIdleConnectionTimeout);
+
+        /**
+         * Forces the client to trust all certificates, even invalid or self signed certificates. This should only ever
+         * be used for testing purposes.
+         *
+         * @param trustAllCertificates Whether to trust all certificates. The default is false and only valid certificates
+         *                             whose trust can be verified via the trust store will be trusted.
+         * @return This builder for method chaining.
+         */
+        Builder trustAllCertificates(Boolean trustAllCertificates);
+
+        /**
+         * Option to enable or disable strict hostname verification when establishing the SSL connection. The default value is
+         * true which means strict hostname verification is enabled by default.
+         *
+         * The host name verification should only be disabled for testing purposes.
+         *
+         * @param useStrictHostNameVerification Whether to use strict hostname verification when establishing the SSL connection.
+         * @return This builder for method chaining.
+         */
+        Builder useStrictHostNameVerification(Boolean useStrictHostNameVerification);
```

Comment:
We should probably kill this option entirely.

Reply:
I was not entirely sure why we are using it. There was some comment about this needing for S3. And the calls to https endpoint fail with SSLPeerUnverifiedException without disabling this option.

Reply:
+1 for killing it

Reply:
This is the thing that allows bucket names with dots. If we use path style it's no longer needed.